### PR TITLE
Deduplicate tags on metrics, events and service_checks for check and dogstatsd

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

De-duplicate tags before pushing to the backend. This avoid pushing hundred of tags if a check is buggy.

### CODE REVIEW

File: checks/__init__.py [hunk 1]
```diff
@@ -584,6 +584,9 @@ def event(self, event):
                 "tags": (optional) list, a list of tags to associate with this event
             }
         """
+        tags = event.get("tags")
+        if tags and isinstance(tags, Iterable):
```
[the remark this answers is not in the document]
Comment (hush-hush) [reply checks/__init__.py]: Events from checks do not go through the `aggregator` but are directly push to the backend from the collector: https://github.com/DataDog/dd-agent/blob/master/emitter.py#L259

[the remark this answers is not in the document]
Comment (masci) [reply checks/__init__.py]: No I mean `isinstance` is to check what?

[the remark this answers is not in the document]
Comment (hush-hush) [reply checks/__init__.py]: To  check that the `tags` entry from the event dict is a iterable (list, set, tuple ...) and can be deduplicate by `set()`.

[the remark this answers is not in the document]
Comment (masci) [reply checks/__init__.py]: Uhm a string would pass the check:
```
In [7]: isinstance("foo", Iterable)
Out[7]: True
```

[the remark this answers is not in the document]
Comment (hush-hush) [reply checks/__init__.py]: hmmm indeed...
Lets remove the check and apply `set` all the time then.

**Pull request merged**